### DISCUSSION TITLE
Are there any plans for getting this upstreamed?

### DISCUSSION THREAD

**Discussion by maxhodak** [protz/thunderbird-conversations#1606] [General]
This seems like it should be core Thunderbird functionality... is there any path to (or interest in?) getting this eventually upstreamed? I'm new to the Thunderbird community and so don't really know how it works, but I definitely see the importance of the last true open source fully-featured local mail/calendar/task client developing into something extremely awesome from the solid foundation it has now. So, I'm just trying to learn more about how the community works and everything fits together.

**Comment by Standard8** [comment]
A lot of add-ons are maintained separately from Thunderbird for various reasons, for example, the functionality might only be useful for a small percentage of users, or the author prefers to keep it separate.

That said, we are investigating the ideas of integration - see the [roadmap discussion](https://github.com/thunderbird-conversations/thunderbird-conversations/discussions/1572) that I commented on yesterday. Even if we don't integrate, there's a lot that would be worth changing in/uplifting to Thunderbird to make Conversations easier to maintain. For example the PDF functionality has now been integrated into the latest Thunderbird development versions and as an added bonus, it has more functionality than Conversations provides.

**Reply by yoshimo** [reply]
Iterating and improving on things becomes easier if you are an addon and not core functionality.
That can be useful if you are up against an unknown, maybe moving target like proprietary protocols or it is just a huge task like an adressbook addon.
